e2e/app/modal/share-modal.ts

```diff
@@ -57,12 +57,18 @@ export default class ShareModal extends Modal {
 
     const typeAndAddUser = async (name: string): Promise<boolean> => {
       const nameWithoutDomain = name.split('@')[0];
-      for (const char of nameWithoutDomain) {
+      // Split string into segments of 4 characters. Type 4 chars at a time, much faster than type 1 char and check.
+      const n = 4;
+      for (let i = 0, charsLength = nameWithoutDomain.length; i < charsLength; i += n) {
+        const chars = nameWithoutDomain.substring(i, i + n);
         const input = await this.waitForSearchBox().asElementHandle();
-        const waitForResponsePromise = this.page.waitForResponse((response) => {
-          return response.url().includes('/userSearch/registered/') && response.request().method() === 'GET';
-        });
-        await input.type(char);
+        const waitForResponsePromise = this.page.waitForResponse(
+          (response) => {
+            return response.url().includes('/userSearch/registered/') && response.request().method() === 'GET';
```

Comment:
typing 4 chars might yield 4 requests, therefore the next time you check this it's possible there's still one running - this doesn't seem too reliable. Do we need this interaction at all?

Reply:
Typing is done without any delay (pause). It should be only one request for all four letters. I am setting `{delay: 0}` explicitly now to make sure that's the case. 
Here is the request example ran on localhost:
```
 [8/23/2021, 14:01:44] - Request issued: OPTIONS https://api-dot-all-of-us-workbench-test.appspot.com/v1/userSearch/registered/pupp

[8/23/2021, 14:00:52] - Request issued: OPTIONS https://api-dot-all-of-us-workbench-test.appspot.com/v1/userSearch/registered/puppetee
```

Reply:
Even if it works now, it seems brittle to me. Can you just wait for network idle instead? This avoids depending on implementation details of which API endpoint gets called and how many times.

Reply:
I understand the concern. I'll keep an eye on this function. I tested it locally a lot, so I hope it will perform well.

`waitForResponse` is actually a better solution than the old solution - wait for Select User dropdown to become visible. It wasn't very reliable.

Wait for network idle can be brittle. For example, wait for idle has started and finished prematurely before any network request `/userSearch`/ is initiated.

I would have to write custom function to do click and wait for network idle in same page because no such functions out-of-box by Puppeteer. Here is a custom function [example](https://stackoverflow.com/questions/54377650/how-can-i-wait-for-network-idle-after-click-on-an-element-in-puppeteer).